src/XeroPHP/Remote/Response.php

```diff
@@ -308,15 +309,28 @@ public function findElementErrors($element, $element_index)
     public function parseXML()
     {
         // There appears to be an issue with some client data returning as unparsable XML
-        // If that happens we want to catch that data and report it
+        // If that happens we want to catch that data and fix it
         try {
             $sxml = new SimpleXMLElement($this->response_body);
         } catch(Throwable $exception) {
-            throw new SimpleXMLException(
-                $exception->getMessage(),
-                $this->response_body,
-                $this->headers[Request::HEADER_CONTENT_LENGTH]
-            );
+            // Try to fix the unparsable XML using Tidy
+            try {
+                $tidy = new Tidy();
+                $config = [
+                    'output-xml'     => true,
+                    'input-xml'     => true,
+                    'numeric-entities' => false,
+                ];
+                $xml = $tidy->repairString($this->response_body, $config);
+                $xml = preg_replace('/[^[:print:]\n\t]/u', '', $xml);
```

Comment:
Any chance of a comment to explain the intention of the match?

Reply:
I would if I knew what it did 😅

Reply:
It looks like this matches any character which is **not** one of the following and removes it.

- a printable character (e.g. ASCII NULL is not a printable character)
- a new line
- a tab

I'd be inclined to remove this line unless we know we need it? E.g. if we know we get non-printable characters or the SimpleXML docs recommend it.

Reply:
@Bittarman thoughts on matt's comment?

Reply:
This line removes the non printable characters which are the issue, without it the bug will remain.
The `'numeric-entities' => false,` part will decode these entities into characters again, and then the non printable will remove the control characters we're seeing.

Reply:
Thanks for clarifying I'll add that explanation as a comment for future reference :)

Reply:
Ah fair enough, in that case we do know we need it!